src/app/component/TranslateOnCursorMove/TranslateOnCursorMove.component.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+import PropTypes from 'prop-types';
+import { createRef, PureComponent } from 'react';
+
+import { ChildrenType } from 'Type/Common';
+import CSS from 'Util/CSS';
+
+import './TranslateOnCursorMove.style';
+
+/** @namespace Component/TranslateOnCursorMove/Component */
+export class TranslateOnCursorMove extends PureComponent {
+    static propTypes = {
+        children: ChildrenType.isRequired,
+        activeImageId: PropTypes.number.isRequired,
+        isMobile: PropTypes.bool.isRequired
+    };
+
+    static defaultProps = {
+    };
+
+    ref = createRef();
+
+    handleMouseMove = ({ pageY: wrapperPageY }) => {
+        const { activeImageId } = this.props;
+
+        // Space from top and bottom to shrink mouse move watch area
+        const paddingY = 90;
+
+        // TODO: need to think how to implement more dynamic way
+        const innerHeight = this.ref.current.children[0].children[1].children[0].children[0]
+            .children[activeImageId].children[0].getBoundingClientRect().height;
```

Comment:
What a monster, lol :'D

You indeed need some way for that, what this even does?